permissions-fixer/src/errors.rs

```diff
@@ -0,0 +1,45 @@
+use std::error::Error;
+use std::fmt;
+use std::fmt::{Debug, Formatter};
+use std::io;
+
+use Failure::*;
+
+#[derive(Eq, PartialEq)]
+pub enum Failure {
+    PathCannotBeEmpty,
+    PathMustBeAbsolute,
+    PathMustBeCanonical,
+    PathMustStartWithPrefix,
+    PathCannotHaveNul,
```

Comment:
Null?

Reply:
The `\0` character is called NUL in ASCII. Anyway this shouldn't happen.